protocol/messenger_pairing_and_syncing.go

```diff
@@ -9,22 +9,53 @@ import (
 	"go.uber.org/zap"
 
 	"github.com/status-im/status-go/eth-node/crypto"
+	"github.com/status-im/status-go/eth-node/types"
 	"github.com/status-im/status-go/images"
 	"github.com/status-im/status-go/protocol/common"
 	"github.com/status-im/status-go/protocol/encryption/multidevice"
 	"github.com/status-im/status-go/protocol/protobuf"
 	"github.com/status-im/status-go/protocol/requests"
 )
 
-func (m *Messenger) EnableAndSyncInstallation(request *requests.EnableAndSyncInstallation) error {
+func (m *Messenger) EnableInstallationAndSync(request *requests.EnableInstallationAndSync) (*MessengerResponse, error) {
 	if err := request.Validate(); err != nil {
-		return err
+		return nil, err
 	}
 	err := m.EnableInstallation(request.InstallationID)
 	if err != nil {
-		return err
+		return nil, err
+	}
+	response, err := m.SendPairInstallation(context.Background(), nil)
+	if err != nil {
+		return nil, err
+	}
+	err = m.SyncDevices(context.Background(), "", "", nil)
+	if err != nil {
+		return nil, err
+	}
+
+	// Delete AC notif
+	notification, err := m.persistence.GetActivityCenterNotificationByID(types.FromHex(request.InstallationID))
+	if err != nil {
+		return nil, err
 	}
-	return m.SyncDevices(context.Background(), "", "", nil)
+
+	if notification != nil {
+		updatedAt := m.GetCurrentTimeInMillis()
+		notification.UpdatedAt = updatedAt
+		notification.Deleted = true
+		// we shouldn't sync deleted notification here,
+		// as the same user on different devices will receive the same message(CommunityCancelRequestToJoin) ?
+		err = m.persistence.DeleteActivityCenterNotificationByID(types.FromHex(request.InstallationID), updatedAt)
+		if err != nil {
+			m.logger.Error("failed to delete notification from Activity Center", zap.Error(err))
+			return nil, err
+		}
+
+		// sending signal to client to remove the activity center notification from UI
+		response.AddActivityCenterNotification(notification)
+	}
```

Comment:
This looks like a duplication from `saveDataAndPrepareResponse`. 
Can you please extract it to a func?